SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialTraceEventDataBuilder.cpp

```diff
@@ -0,0 +1,277 @@
+
+#include "Math/UnrealMathUtility.h"
+
+DEFINE_LOG_CATEGORY(FSpatialTraceEventDataBuilderLog);
+
+namespace SpatialGDK
+{
+// ---- FSpatialTraceEventName ----
+
+const char* FSpatialTraceEventName::PushRPCEventName = GDK_EVENT_NAMESPACE "push_rpc";
+const char* FSpatialTraceEventName::ComponentUpdateEventName = GDK_EVENT_NAMESPACE "component_update";
+const char* FSpatialTraceEventName::SendRetireEntityEventName = GDK_EVENT_NAMESPACE "send_retire_entity";
+const char* FSpatialTraceEventName::PropertyChangedEventName = GDK_EVENT_NAMESPACE "property_changed";
+const char* FSpatialTraceEventName::SendPropertyUpdateEventName = GDK_EVENT_NAMESPACE "send_property_update";
+const char* FSpatialTraceEventName::SendCreateEntityEventName = GDK_EVENT_NAMESPACE "send_create_entity";
+const char* FSpatialTraceEventName::ReceiveCreateEntitySuccessEventName = GDK_EVENT_NAMESPACE "receive_create_entity_success";
+const char* FSpatialTraceEventName::ReceivePropertyUpdateEventName = GDK_EVENT_NAMESPACE "receive_property_update";
+const char* FSpatialTraceEventName::AuthorityIntentUpdateEventName = GDK_EVENT_NAMESPACE "authority_intent_update";
+const char* FSpatialTraceEventName::ReceiveCommandResponseEventName = GDK_EVENT_NAMESPACE "receive_command_response";
+const char* FSpatialTraceEventName::ReceiveCommandRequestEventName = GDK_EVENT_NAMESPACE "receive_command_request";
+const char* FSpatialTraceEventName::ApplyRPCEventName = GDK_EVENT_NAMESPACE "apply_rpc";
+const char* FSpatialTraceEventName::SendRPCEventName = GDK_EVENT_NAMESPACE "send_rpc";
+const char* FSpatialTraceEventName::QueueRPCEventName = GDK_EVENT_NAMESPACE "queue_rpc";
+const char* FSpatialTraceEventName::ReceieveRPCEventName = GDK_EVENT_NAMESPACE "receive_rpc";
+const char* FSpatialTraceEventName::MergeSendRPCEventName = GDK_EVENT_NAMESPACE "merge_send_rpcs";
+const char* FSpatialTraceEventName::ApplyCrossServerRPCEventName = GDK_EVENT_NAMESPACE "apply_cross_server_rpc";
+const char* FSpatialTraceEventName::SendCommandResponseEventName = GDK_EVENT_NAMESPACE "send_command_response";
+const char* FSpatialTraceEventName::SendCommandRequestEventName = GDK_EVENT_NAMESPACE "send_command_request";
+const char* FSpatialTraceEventName::SendCrossServerRPCEventName = GDK_EVENT_NAMESPACE "send_cross_server_rpc";
+const char* FSpatialTraceEventName::ReceiveCrossServerRPCEventName = GDK_EVENT_NAMESPACE "receive_cross_server_rpc";
+const char* FSpatialTraceEventName::MergePropertyUpdateEventName = GDK_EVENT_NAMESPACE "merge_property_update";
+const char* FSpatialTraceEventName::ReceiveCreateEntityEventName = GDK_EVENT_NAMESPACE "receive_create_entity";
+const char* FSpatialTraceEventName::ReceiveRemoveEntityEventName = GDK_EVENT_NAMESPACE "receive_remove_entity";
+const char* FSpatialTraceEventName::AuthorityChangeEventName = GDK_EVENT_NAMESPACE "authority_change";
+
+// ---- FStringCache ----
+
+int32 FSpatialTraceEventDataBuilder::FStringCache::CombineStrings(const char* A, const char* B)
```

Comment:
I wonder if we could service this functionality by using AddString but subtracting 1 from NextIndex to start at the end of the previous string - maybe with a check if NextIndex = 0?

Reply:
If AddString is internal we can ensure the usage is always good

Reply:
Perhaps we could pass in the start index and do something like:
int32 Next = AddString("x");
AddString("x", Next-1);

and default to -1 meaning NextIndex?

Reply:
Believe I have added a valid solutions to this